lib/linebuffer.js

```diff
@@ -130,7 +130,7 @@ const flush = () => {
                     level: config.GZIP_COMPRESS_LEVEL
                 }, (err, out) => {
                     config.req({
-                        url: config.LOGDNA_LOGURL + config.LOGDNA_LOGENDPOINT
+                        url: utils.buildUrl(config.LOGDNA_LOGHOST, config.LOGDNA_LOGPORT, config.LOGDNA_LOGSSL, config.LOGDNA_LOGENDPOINT)
```

Comment:
I see a lot of these utils.buildUrl(), using the same parameters over and over again, could we not add `url` to request.defaults? or does request not allow you to have a default `url`?

Reply:
I have done it - even moved `method` and `headers` too - and didn't know that you can override already-defaulted `request` fields thru [`request.defaults({...}).defaults({...})`](https://github.com/request/request#requestdefaultsoptions).